werkit/compute/graph/_value_types.py

```diff
@@ -0,0 +1,6 @@
+SUPPORTED_TYPES = ["Number", "Point", "Measurement"]
+
+
+def assert_valid_value_type(value_type: str) -> None:
```

Comment:
Hmm, it's an interesting idea. There are a few related concerns here:
- Declaring the type in the decorator
- Returning the computed value at runtime
- Validating the result
- Serializing the result

Making up new result classes which descend from a BaseResult class and returning instances of those objects would address all these concerns. Though, it may be less convenient for the implementer. Having to wrap all return values into these classes would be less convenient than returning e.g. a numpy array, or a number.

Another option would be to use classes as you suggest, to declare, serialize, and validate the results, but accept return types which are either instances of the return type class, or values which can be coerced to that return type. The value class could be responsible for implementing this coercion which feels pretty clean.

I like this approach. While it prioritizes the concerns of the implementer by making the easy thing easy, it also does a better job addressing the concerns you're raising around the type declarations. I think it is also a very nice way to implement validation and serialization.